schema2Markdown.js

```diff
@@ -0,0 +1,255 @@
+}
+
+function hasProperties(schema) {
+    return schema.properties != null;
+}
+
+// Function to generate Markdown content from JSON schema
+function generateObjectMD(schema, title, schemaFolderName, filePath, version) {
+
+    const objectName = schema.title
+
+    if (schema.title != null) {
+        title = schema.title;
+    }
+
+    let markdownContent = `# ${title}\n\n`;
+
+    if (schema.description != null) {
+        markdownContent += `${escape(schema.description)}\n\n`; 
+    }
+
+    if (schema.type) {
+        markdownContent += renderType(schema.type);
+    }
+    if (schema.enum) {
+        markdownContent += renderEnum(schema.enum);
+    }
+
+    if (hasAllOf(schema.allOf) || hasProperties(schema)) {
+        // Extract properties, required fields, and $ref from the first allOf object
+        let root = schema;
+        if (hasAllOf(schema.allOf)) {
+            root = schema.allOf[0];
+        }
+        
+        const properties = root.properties;
+        const required = root.required;
+        const ref = root.$ref;
+
+        markdownContent += `## Properties\n\n`; 
+
+        for (const [propertyName, propertyDetails] of Object.entries(properties)) {
+            markdownContent += processProperty(propertyName, propertyDetails, schema.examples);
+        }
+
+        // show required properties
+        if (required && required.length > 0) {
+            markdownContent += `### Required Properties:\n\n`;
+            // for each required property show the name
+            required.forEach((propertyName) => {
+                markdownContent += `* \`${propertyName}\`\n`;
+            });
+            markdownContent += '\n';
+        } 
+
+        if (ref) {
+            markdownContent += `ref: ${ref}\n\n`;
+        }
+
+        if (schema.examples) {
+            markdownContent += `## Examples\n\n`;
+            markdownContent += '```json\n';
+            markdownContent += JSON.stringify(schema.examples, null, 2);
+            markdownContent += '\n```\n\n';
+        }
+        const url = filePath.replace("schemas/", `https://github.com/finos/FDC3/tree/master/schemas/`);
+        markdownContent += `Generated from ${url}`;
+        
+    }
+
+    const frontMatter = generateFrontMatter(objectName, schema.description);
+
+    const outputFileName = `./website/versioned_docs/version-${version}/${schemaFolderName}/schemas/${title.replace(/\s+/g, '')}.md`;
+
+    fse.outputFileSync(outputFileName, `---\n${yaml.dump(frontMatter)}\n---\n\n${markdownContent}`);
+
+    // objectName must not contain any spaces
+    if (objectName != null) {
+        return schemaFolderName + '/schemas/' + objectName.replace(/\s+/g, '');
+    }
+}
+
+function escape(text) {
+    return text.replace(/>/g, '\\>');
+}
+
+function generateFrontMatter(title, description) {
+    return {
+        title: `${title} Schema`,
+        description: description,
+        sidebar_label: title + ' Schema',
+    };
+}
```

Comment:
No need to add 'Schema'
```suggestion
function generateFrontMatter(title, description) {
    return {
        title: title,
        description: description,
        sidebar_label: title,
    };
}
```